### PR TITLE
Improve token search performance

### PR DESCRIPTION
**Pull request by sssmi**
# Description

- Removed extraneous loading prop
- First keypress on search input caused UI to freeze for ~3 seconds. Removed nested loops from `mapTokenListTokens`.

## Type of change

- [x] Code refactor / cleanup

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

1. Go to home page
2. Click "Filter by token" button
3. Start typing to token search
4. Token list should update immediately

### First keypress event run time dropped from 3.38s to 43ms
| Before | Now |
| ------ | ---- |
| Screen Shot 2022-07-01 at 12 39 44   | Screen Shot 2022-07-01 at 12 37 41  |

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas
- [x] If package-lock.json has changes, it was intentional.
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

**Comment by pkattera** [comment]
@sssmi In terms of the performance improvement – this is outstanding!

File: src/providers/tokens.provider.ts [hunk 1]
```diff
@@ -249,22 +249,20 @@ export default {
      * Create token map from a token list tokens array.
      */
     function mapTokenListTokens(tokenLists: TokenList[]): TokenInfoMap {
-      const tokensMap = {};
       const tokens = [...tokenLists].map(list => list.tokens).flat();
 
-      tokens.forEach(token => {
+      const tokensMap = tokens.reduce<TokenInfoMap>((acc, token) => {
```
Comment (alter-eggo) [inline src/providers/tokens.provider.ts:254]: 👍

**Review by alter-eggo — COMMENTED** [review]
nice work!

**Review by alter-eggo — APPROVED** [review]

**Review by timjrobinson — APPROVED** [review]
Nice work! From n^2 to n lookup time.

**Pull request merged**